src/main/java/org/cmps/tetrahedron/utils/DataReader.java

```diff
@@ -42,17 +52,22 @@ public static List<float[][]> readIndexesAndConvertToFaces(File indicesMatrix,
 
         try (Scanner fid = new Scanner(indicesMatrix)) {
             List<float[][]> faces = new ArrayList<>();
-
-            while (fid.hasNext()) {
-                // reading index of a tetrahedron
-                if (WITH_INDICES) {
-                    fid.nextInt();
+            while (fid.hasNextLine()) {
+                String[] elements = fid.nextLine().trim().split("\\s+");
+                float[] vertex1, vertex2, vertex3, vertex4;
+
+                if (elements.length == FACE_WITH_INDICES) {
+                    vertex1 = verticesCoordinates.get(Integer.parseInt(elements[1]));
```

Comment:
are there any ways not to duplicate the code inside the if-else operator?

Reply:
I've tried, but so far I couldn't done it.